adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -0,0 +1,221 @@
+                       completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler {
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block  GADMediationNativeLoadCompletionHandler originalCompletionHandler = [completionHandler copy];
+    _loadCompletionHandler = ^id<GADMediationNativeAdEventDelegate> (_Nullable id<GADMediationNativeAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationNativeAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";
+    if (!placementId.length) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorInvalidServerParameters,
+                                                                      [NSString stringWithFormat:@"%@ cannot be nil.",GADMAdapterPanglePlacementID]
+                                                                      );
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    
+    BUAdSlot *slot = [[BUAdSlot alloc] init];
+    slot.ID = placementId;
+    slot.AdType = BUAdSlotAdTypeFeed;
+    
+    _nativeAd = [[BUNativeAd alloc]initWithSlot:slot];
+    _nativeAd.delegate = self;
+    [_nativeAd setAdMarkup:adConfiguration.bidResponse];
+}
+
+- (BUNativeAdRelatedView *)getRelatedView {
+    if (!_relatedView) {
+        _relatedView = [[BUNativeAdRelatedView alloc] init];
+    }
+    return _relatedView;
+}
+
+#pragma mark GADMediationNativeAd
+
+- (UIView *)mediaView {
+  if (!_mediaView) {
+    BUVideoAdView *videoView = [self getRelatedView].videoAdView;
+    _mediaView = videoView;
+  }
+  return _mediaView;
+}
+
+- (UIView *)adChoicesView {
+    return [self getRelatedView].logoADImageView;
+}
+
+- (NSString *)headline {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSString *)body {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdDescription;
+    }
+    return nil;
+}
+
+- (NSString *)callToAction {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.buttonText;
+    }
+    return nil;
+}
+
+- (NSDecimalNumber *)starRating {
+    return [NSDecimalNumber decimalNumberWithString:[NSString stringWithFormat:@"%ld",(long)_nativeAd.data.score]];
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+- (NSString *)advertiser {
+    if (_nativeAd && _nativeAd.data) {
+        return _nativeAd.data.AdTitle;
+    }
+    return nil;
+}
+
+- (NSDictionary<NSString *,id> *)extraAssets {
+    if (_nativeAd) {
+        return @{BUDNativeAdTranslateKey:_nativeAd};
```

Comment:
Nit, if this is the only place that uses `bu_nativeAd`, we can just have it here and don't need an extra const.